packages/abi-coder/src/constants.ts

```diff
@@ -13,6 +13,12 @@ export const MAX_INPUTS = 255;
 export const ASSET_ID_LEN = BYTES_32;
 export const CONTRACT_ID_LEN = BYTES_32;
 
+export const calculateVmTxMemory = ({ maxInputs }: { maxInputs: number }) =>
+  BYTES_32 + // Tx ID
+  WORD_SIZE + // Tx size
+  // Asset ID/Balance coin input pairs
+  maxInputs * (ASSET_ID_LEN + WORD_SIZE);
+
```

Comment:
As requested in https://github.com/FuelLabs/fuels-ts/issues/1050, we don't need to use  hard-coded value for `MAX_INPUTS` anymore. We can safely use a value fetched from the chain, passed down from a provider object.